### PR TITLE
show "open for both" color on receiving/distributing filter buttons

### PR DESCRIPTION
**Pull request by maxine**
In response to conversation in Slack with @comeoneileen @jasonthibeault, added the orange "open for both" color alongside the "open for receiving" and "open for distributing" colors in the filtering / legend area. Also hid the "open for both" filtering option. 

Currently: 
Screen Shot 2020-06-03 at 5 52 55 PM

Previously:
image

### CODE REVIEW

**Comment by mc-funk** [comment]
I super appreciate this work but I honestly find this more confusing... "open for both" was pretty clear about what the colors meant and was more consistent with a typical legend layout.  That could just be me, though, I'd be curious to know what @comeoneileen and others think

**Comment by hinzed1127** [comment]
This could be a good case for the "Both" checkbox to have a mixed state like described [here] (https://css-tricks.com/indeterminate-checkboxes/)

Basically, have a nested list like this:

- [ ] Receiving/Distributing Donations
    - [ ] Receiving Donations
    - [ ] Distributing Donations

by default have all three checked. if the top level on is unchecked, uncheck all 3. If only one of the two sub-level checkboxes is selected, put the top level checkbox in the indeterminate state. between the mixed state and the indented hierarchy, that _should_ help distinguish the different meanings, imo.

curious how that sounds

**Comment by jasonthibeault** [comment]
@hinzed1127 Unfortunately that would not work for this use case: orange is "receives AND distributes", not "OR". It's a way of distinguishing places that do both, because the pin can't be both light-blue AND dark-blue

**Comment by jdalt** [comment]
Making the legend and filter the same are tricky since not everything is 1-1. We might also want to consider alternative ways the pins could work.

**Comment by hinzed1127** [comment]
@jasonthibeault  that's a good point, I was only thinking in terms of the checkboxes and not the associated colors. I guess you could still enforce that logic to the colors and have the mixed checkbox state, idk if others thing that makes things clearer or not.

**Comment by kmarekspartz** [comment]
Perhaps the colors should be changed to help with this? Orange isn't clear to me as both light and dark blue. 

https://colorbrewer2.org/#type=diverging&scheme=RdYlBu&n=5

#d7191c -> confirmed closed
#fdae61 -> unknown
#ffffbf -> open for receiving
#abd9e9 -> open for distributing
#2c7bb6 -> open for both

**Comment by maxine** [comment]
Screen Shot 2020-06-03 at 8 50 41 PM
Refactored colors based on [Coloring for Colorblindness](https://davidmathlogic.com/colorblind/#%23332288-%23117733-%2344AA99-%2388CCEE-%23DDCC77-%23CC6677-%23AA4499-%23882255) by David Nichols (h/t @samanpwbb) and to hopefully make the mapping of the colors more intuitive. 
This still doesn't solve @mc-funk's note: "this part of the UI isn’t just filters. It’s filters and the color key for the pins... So people need to be able to see 'orange [now maybe green] means both'"

**Comment by jdalt** [comment]
@maxine are you familiar with this contrast accessibility validator that @cwood821 pointed out in https://github.com/Twin-Cities-Mutual-Aid/twin-cities-aid-distribution-locations/issues/59 ?

**Comment by jdalt** [comment]
@maxine You also might want to coordinate with @samanpwbb on https://github.com/Twin-Cities-Mutual-Aid/twin-cities-aid-distribution-locations/issues/57#issuecomment-638561918

There seems to be a some overlap of ideas and work.

**Comment by samanpwbb** [comment]
> @maxine You also might want to coordinate with @samanpwbb on [#57 (comment)](https://github.com/Twin-Cities-Mutual-Aid/twin-cities-aid-distribution-locations/issues/57#issuecomment-638561918)

We coordinated – I based my suggestions off of this PR :)

File: filter.js [hunk 1]
```diff
@@ -61,7 +60,12 @@ class Filter {
 
 
     const filters = this.statusOptions.map(s => {
-      return `<li class='filter-item'><input class="filter" type="checkbox" id="filter-${s.name}" value="${s.name}" checked><span class="legend--item--swatch" style="background-color: ${s.accessibleColor}"></span><label for="filter-${s.name}">${s.label}</label></li>`
+      /** Open for receiving / distributing donations should have two colors */
+      if (s.accessibleColor === "#2c7bb6" || s.accessibleColor === "#abd9e9") {
+        return `<li class='filter-item'><input class="filter" type="checkbox" id="filter-${s.name}" value="${s.name}" checked><span class="legend--item--swatch" style="background-color: ${s.accessibleColor}"></span>/<span class="legend--item--swatch" style="background-color: #fdae61"></span><label for="filter-${s.name}">${s.label}</label></li>`
+      } else {
+        return `<li class='filter-item'><input class="filter" type="checkbox" id="filter-${s.name}" value="${s.name}" checked><span class="legend--item--swatch" style="background-color: ${s.accessibleColor}"></span><label for="filter-${s.name}">${s.label}</label></li>`
+      }
```
Comment (kmarekspartz) [inline filter.js:68]: There's a lot duplicated in both clauses here. Can we extract a variable and have the duplicated string at the end?

File: script.js [hunk 2]
```diff
@@ -268,6 +268,16 @@ const key = document.getElementById('key')
 statusOptions.forEach(s => {
   const el = document.createElement('div');
   el.classList = ['legend--item'];
-  el.innerHTML = `<span class="legend--item--swatch" style="background-color: ${s.accessibleColor}"></span>${s.label}`
+  if (s.accessibleColor === "#2c7bb6" || s.accessibleColor === "#abd9e9") {
+    el.innerHTML = `<span class="legend--item--swatch" style="background-color: ${s.accessibleColor}"></span> / <span class="legend--item--swatch" style="background-color: #fdae61"></span>${s.label}`
+  } else {
+    el.innerHTML = `<span class="legend--item--swatch" style="background-color: ${s.accessibleColor}"></span>${s.label}`
+  }
```
Comment (kmarekspartz) [inline script.js:275]: Same idea here, though here it might be extracting a helper method.

File: script.js [hunk 3]
```diff
@@ -268,6 +268,16 @@ const key = document.getElementById('key')
...
   key.append(el)
 })
+
+/*
+  if ($item.dataset.id === "filter-recieving" || $item.dataset.id === "filter-distributing") {
+    const span = `<span title="${status.id}" class="status location-list--indicator" style="background-color: ${status.accessibleColor};">${status.id}</span>`;
+  };
+*/
```
Comment (kmarekspartz) [inline script.js:283]: Do we need this code any more?

**Comment by maxine** [comment]
Thanks for everyone's feedback on this! Closing this PR after talking with folks in Slack about keeping the colors the same so people who have been using the map don’t have to re-learn the color meanings + merging in #66 to fix the original checkbox issue.

**Pull request closed without merging**